aspnetcore/fundamentals/websockets.md

```diff
@@ -92,6 +92,32 @@ The WebSocket closed exception can also happen when returning too soon from an a
 
 Never use `Task.Wait`, `Task.Result`, or similar blocking calls to wait for the socket to complete, as that can cause serious threading issues. Always use `await`.
 
+::: moniker range=">= aspnetcore-6.0"
+
+### Using Compression
+
+> [!WARNING]
+> Enabling compression over encrypted connections makes the application subject to CRIME/BREACH type attacks.
+> If you are sending sensitive information consider not enabling compression, or make sure you use [WebSocketMessageFlags.DisableCompression](TODO) when calling [WebSocket.SendAsync](TODO).
+> This applies to both sides of the WebSocket, it's worth noting that the WebSockets API in the browser doesn't have configuration for disabling compression per send. 
+
+If compression of messages over WebSockets is desired then the accept code needs to specify that it allows compression as follows:
+```csharp
+using (WebSocket webSocket = await context.WebSockets.AcceptWebSocketAsync(
+    new WebSocketAcceptContext() { DangerousEnableCompression = true }))
+{
+}
+````
+
+Additionally, there are the [ServerMaxWindowBits](TODO) and [DisableServerContextTakeover](TODO) advanced options on [WebSocketAcceptContext](dotnet/api/microsoft.aspnetcore.http.websocketacceptcontext) that control how the compression works.
```

Comment:
```suggestion
Additionally, `ServerMaxWindowBits` and `DisableServerContextTakeover` advanced options of <xref:Microsoft.AspNetCore.Http.WebSocketAcceptContext> control how the compression works.
```

Reply:
This sounds off for some reason

Reply:
It's the "of" bit. It could go ....

```suggestion
Additionally, <xref:Microsoft.AspNetCore.Http.WebSocketAcceptContext> advanced options, `ServerMaxWindowBits` and `DisableServerContextTakeover`, control how the compression works.
```

Reply:
Also, this is a possibility ...

```suggestion
Additionally, `WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` control how the compression works.
```

Reply:
> This sounds off for some reason

agreed. Original is fine if they are really advanced, meaning most users won't use them.

Reply:
```suggestion
[ServerMaxWindowBits](TODO) and [DisableServerContextTakeover](TODO) are advanced options on [WebSocketAcceptContext](dotnet/api/microsoft.aspnetcore.http.websocketacceptcontext) that control how the compression works.
```

Reply:
Additionally is dead wood.

Reply:
The reason I suggested a change for the original was to avoid the state verb. I think my last suggestion is probably the best of the litter. 🐱

Reply:
```suggestion
`WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` control how the compression works.
```

... using `?displayProperty=nameWithType` later when these to go API links.

Reply:
```suggestion
`WebSocketAcceptContext.ServerMaxWindowBits` and `WebSocketAcceptContext.DisableServerContextTakeover` are advanced options that control how the compression works.
```